core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -5919,6 +5921,30 @@ private Object coerce(Type sourceType, Object value, Type targetType)
         }
     }
 
+    private static RefreshType getRefreshType(Query query)
+    {
+        if (query.getQueryBody() instanceof QuerySpecification querySpecification) {
+            if (querySpecification.getGroupBy().isEmpty()
+                    && querySpecification.getWindows().isEmpty()
+                    && querySpecification.getFrom().stream().allMatch(from -> from instanceof Table)
+                    && !querySpecification.getSelect().isDistinct()
+                    && getSelectFunctionNames(querySpecification).isEmpty()
```

Comment:
Why function filtering is needed? BTW: Even if there are no functions here, they might be added by column masking/row filtering.